### PR TITLE
Do not use patching viewer in make_napari_viewer fixture teardown

### PR DESCRIPTION
**Pull request by Czaki**
# References and relevant issues

fixing such errors

```
napari/_qt/_tests/test_qt_viewer.py::test_add_layer[Image-data2-4] PASSED [ 7/75]Fatal Python error: Segmentation fault

Current thread 0x00000001df171000 (most recent call first):
  File "/Users/sobolp/Documents/dev/napari/napari/utils/_testsupport.py", line 269 in make_napari_viewer
````

# Description

As monkeypatch contains reverse patching steep, it looks like it causing segfaults on Qt6 (it happens more often on PySide6, but also on 3.12+PyQt6.

### CODE REVIEW

**Comment by psobolewskiPhD** [comment]
Do you want to bump the pyside6 contraint to 6.4.3 to see that this closes https://github.com/napari/napari/issues/5657 ?
(blocked since https://github.com/napari/napari/pull/5683 )

**Comment by psobolewskiPhD** [comment]
I pushed a fix, you forgot to let tox use 6.4.3 :smile:

**Comment by Czaki** [comment]
Next ugly code...

**Comment by psobolewskiPhD** [comment]
I re-ran it here's the new seg fault:
https://github.com/napari/napari/actions/runs/8285842683/job/22674979082

**Comment by Czaki** [comment]
Yes. The general problem with pyside(2/6) is that they overwrite python `object` with its own problematic one.

**Comment by Czaki** [comment]
well. More bugs..

**Comment by psobolewskiPhD** [comment]
whack-a-mole

File: napari/_qt/qt_main_window.py [hunk 1]
```diff
@@ -1611,13 +1611,9 @@ def _teardown(self):
 
     def close(self):
         """Close the viewer window and cleanup sub-widgets."""
-        # Someone is closing us twice? Only try to delete self._qt_window
-        # if we still have one.
-        if hasattr(self, '_qt_window'):
-            self._teardown()
-            self._qt_viewer.close()
-            self._qt_window.close()
-            del self._qt_window
```
Comment (psobolewskiPhD) [inline napari/_qt/qt_main_window.py:1634]: worth putting this `del` back?

Comment (Czaki) [reply napari/_qt/qt_main_window.py:1634]: no. This `del` causes segfaults. I told that pyside objects are crap

Comment (Czaki) [reply napari/_qt/qt_main_window.py:1634]: oh. crap is worst than expected...

Comment (psobolewskiPhD) [reply napari/_qt/qt_main_window.py:1634]: 😬

Comment (psobolewskiPhD) [reply napari/_qt/qt_main_window.py:1634]: maybe push these changes to the 6.5.1 fix branch and see if >6.5 helps?

Comment (Czaki) [reply napari/_qt/qt_main_window.py:1634]: again lambda in face...

**Comment by Czaki** [comment]
could we nuke PySide?

```

    File "/home/runner/work/napari/napari/.tox/py311-linux-pyside6-no_cov/lib/python3.11/site-packages/_pytest/unraisableexception.py", line 59 in __exit__
    File "/home/runner/work/napari/napari/.tox/py311-linux-pyside6-no_cov/lib/python3.11/site-packages/_pytest/unraisableexception.py", line 63 in unraisable_exception_runtest_hook
    File "/home/runner/work/napari/napari/.tox/py311-linux-pyside6-no_cov/lib/python3.11/site-packages/_pytest/unraisableexception.py", line 95 in pytest_runtest_teardown
```

Where this line contains... `del self.unraisable`

File: napari/utils/_testsupport.py [hunk 2]
```diff
@@ -267,13 +269,10 @@ def actual_factory(
...
     for viewer in viewers:
-        if hasattr(viewer.window, '_qt_window'):
-            with patch.object(
-                viewer.window._qt_window, '_save_current_window_settings'
-            ):
-                viewer.close()
-        else:
-            viewer.close()
+        viewer.close()
+    del viewer
```
Comment (psobolewskiPhD) [inline napari/utils/_testsupport.py]: Do you think this `del` could be a problem?

**Comment by Czaki** [comment]
@jni @psobolewskiPhD 
I do not know what to do with this PR. It now segfaults on cleaning of `AutoScaleButtons` when cleaning dict. Even when I have removed the lambda connection from it. Checking C stacktrace it suggests that PySide has already removed some object that is referred by python. 

Maybe it should wait until 6.5 (6.6) fixes will be merged?

**Comment by psobolewskiPhD** [comment]
> Maybe it should wait until 6.5 (6.6) fixes will be merged?

You mean this PR to deal with the swapping of class inheritance order?
https://github.com/napari/napari/pull/6062

I mean I think the fix there is correct as far as i can understand the pyside ticket and that approach is used in other places. That PR has gotten gnarly with the constraints and I think you pushed something to try and fix the set faults, but I think we could back it up to just the fix for inheritance order and merge it with the understanding that CI will not be green, but it could make life easier for this PR to fix the seg faults?
As long as non-pyside CI is green I'd be in favor.

**Comment by Czaki** [comment]
Meh. Maybe will back again in the future, but with new PR.

**Pull request closed without merging**